examples/bundles/bundle_with_multiple_entries/main.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <assert.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <string>
+#include <vector>
+
+#include "testBundle.h"
+
+//===----------------------------------------------------------------------===//
+//                 Wrapper code for executing a bundle
+//===----------------------------------------------------------------------===//
+/// Find in the bundle's symbol table a weight variable whose name starts with
+/// \p name.
+const SymbolTableEntry *getWeightVar(const BundleConfig &config,
+                                     const char *name) {
+  for (unsigned i = 0, e = config.numSymbols; i < e; ++i) {
+    if (!strncmp(config.symbolTable[i].name, name, strlen(name))) {
+      return &config.symbolTable[i];
+    }
+  }
+  return nullptr;
+}
+
+/// Find in the bundle's symbol table a mutable weight variable whose name
+/// starts with \p name.
+const SymbolTableEntry &getMutableWeightVar(const BundleConfig &config,
+                                            const char *name) {
+  const SymbolTableEntry *mutableWeightVar = getWeightVar(config, name);
+  if (!mutableWeightVar) {
+    printf("Expected to find variable '%s'\n", name);
+  }
+  assert(mutableWeightVar && "Expected to find a mutable weight variable");
+  assert(mutableWeightVar->kind != 0 &&
+         "Weight variable is expected to be mutable");
+  return *mutableWeightVar;
+}
+
+/// Allocate an aligned block of memory.
+void *alignedAlloc(const BundleConfig &config, size_t size) {
+  void *ptr;
+  // Properly align the memory region.
+  int res = posix_memalign(&ptr, config.alignment, size);
+  assert(res == 0 && "posix_memalign failed");
+  assert((size_t)ptr % config.alignment == 0 && "Wrong alignment");
+  memset(ptr, 0, size);
+  (void)res;
+  return ptr;
+}
+
+/// Initialize the constant weights memory block by loading the weights from the
+/// weights file.
+static uint8_t *initConstantWeights(const char *weightsFileName,
+                                    const BundleConfig &config) {
+  // Load weights.
+  FILE *weightsFile = fopen(weightsFileName, "rb");
```

Comment:
Why not use LLVM's file open/write wrapper? i.e. raw_fd_ostream

Reply:
IIRC, all of our bundles examples do not use any of LLVM APIs as they are examples of "standalone" apps.